rules/legacy_xcodeproj.bzl

```diff
@@ -1054,7 +1054,7 @@ def _xcodeproj_impl(ctx):
     (xcodeproj_targets_by_name, xcodeproj_schemes_by_name) = _populate_xcodeproj_targets_and_schemes(ctx, targets, src_dot_dots, all_transitive_targets)
 
     project_file_groups = [
-        {"path": paths.join(src_dot_dots, f.short_path), "optional": True}
```

Comment:
Any downsides to not setting `optional` here?

Reply:
I don't think we have the option to set `optional` here. 

`fileGroups` should be a list of paths to add to the root of the project. If we include a list of `(path: XX, optional: XX)` like this, the additional files, i.e., the `fileGroups` will not be included in the generated xcode project.
```
fileGroups:
  - path: "/Users/tawang/Downloads/xcodegen-localized/rules-ios/objc/BUILD.bazel"
    optional: true
```

Reply:
I see so this was just broken eh? I don't have the context for when this was added but maybe who did it just replicated the pattern for sources (path + optional set). From the docs this seems like the correct impl indeed. Nice catch!

Reply:
yeap this is broken right now.

sorry I missed Jerry's general comment for the glob patten. I wonder if adding a `filegroup` with`glob(include, exclude=[]` patten, then add this filegroup to `additional_files` is enough for doing the similar optional work here?